specification/app/resource-manager/Microsoft.App/preview/2024-02-02-preview/ContainerAppsPatches.json

```diff
@@ -0,0 +1,563 @@
+          "description": "Properties that describes current states of the patch resource.",
+          "x-ms-mutability": [
+            "read",
+            "update"
+          ]
+        }
+      },
+      "description": "Container App Patch",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "PatchProvisioningState": {
+      "type": "string",
+      "description": "Resource instance provisioning state.",
+      "enum": [
+        "Succeeded",
+        "Failed",
+        "Canceled"
+      ],
+      "x-ms-enum": {
+        "name": "PatchProvisioningState",
+        "modelAsString": true
+      },
+      "readOnly": true
+    },
+    "PatchApplyStatus": {
+      "type": "string",
+      "description": "The status of the patch once it has been provisioned",
+      "enum": [
+        "NotStarted",
+        "RebaseInProgress",
+        "CreatingRevision",
+        "Succeeded",
+        "Canceled",
+        "RebaseFailed",
+        "RevisionCreationFailed",
+        "ImagePushPullFailed",
+        "ManuallySkipped"
+      ],
+      "x-ms-enum": {
+        "name": "PatchApplyStatus",
+        "modelAsString": true
+      },
+      "readOnly": true
+    },
+    "DetectionStatus": {
+      "type": "string",
+      "description": "The status of the patch detection.",
+      "enum": [
+        "Succeeded",
+        "RegistryLoginFailed",
+        "Failed"
+      ],
+      "x-ms-enum": {
+        "name": "DetectionStatus",
+        "modelAsString": true
+      },
+      "readOnly": true
+    },
+    "PatchDetails": {
+      "description": "The detailed info of patch operation performing when applying a patch.",
+      "type": "object",
+      "required": [
+        "targetContainerName",
+        "targetImage",
+        "lastDetectionTime",
+        "detectionStatus"
+      ],
+      "properties": {
+        "targetContainerName": {
+          "type": "string",
+          "description": "The name of the target container for the patch.",
+          "readOnly": true
+        },
+        "targetImage": {
+          "type": "string",
+          "description": "The name of the target image for the patch.",
+          "readOnly": true
+        },
+        "lastDetectionTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The UTC timestamp that describes the latest detection was done.",
+          "readOnly": true
+        },
+        "detectionStatus": {
+          "$ref": "#/definitions/DetectionStatus",
+          "description": "The status of the patch detection.",
+          "readOnly": true
+        },
+        "newImageName": {
+          "type": "string",
+          "description": "The name of the new image created by the patch.",
+          "readOnly": true
+        },
+        "newLayer": {
+          "type": "object",
+          "properties": {
```

Comment:
Consider using `x-ms-client-flatten` for nested properties: https://github.com/Azure/azure-openapi-validator/blob/main/docs/avoid-nested-properties.md

Reply:
I don't think you want to use flatten here - your newLayer and oldLayer property names will collide if you do.